server/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -73,6 +74,22 @@ public FsBlobContainer(FsBlobStore blobStore, BlobPath blobPath, Path path) {
         return listBlobsByPrefix(null);
     }
 
+    @Override
+    public Map<String, BlobContainer> children() throws IOException {
+        // If we get duplicate files we should just take the last entry
```

Comment:
why would we expect duplicates?

Reply:
Good question, for some reason I just mindlessly copied that comment from the list by prefix method. It makes 0 sense imo, I'll just remove it.